site/docs/contributing.md

```diff
@@ -0,0 +1,267 @@
+---
+title: Contributing to promptfoo
+sidebar_label: Contributing
+---
+
+We welcome contributions from the community to help make promptfoo better. This guide will help you get started. If you have any questions, please reach out to us on [Discord](https://discord.gg/gHPS9jjfbs) or through a [GitHub issue](https://github.com/promptfoo/promptfoo/issues/new).
+
+## Project Overview
+
+promptfoo is an MIT licensed tool for testing and evaluating LLM apps.
+
+We particularly welcome contributions in the following areas:
+
+- Bug fixes
+- Documentation updates, including examples and guides
+- Updates to providers including new models, new capabilities (tool use, function calling, JSON mode, file uploads, etc.)
+- Features that improve the user experience of promptfoo, especially relating to RAGs, Agents, and synthetic data generation.
+
+## Getting Started
+
+1. Fork the repository on GitHub.
+2. Clone your fork locally:
+
+   ```sh
+   git clone https://github.com/[your-username]/promptfoo.git
+   cd promptfoo
+   ```
+
+3. Set up your development environment:
+
+   ```sh
+   # We recommend using the Node.js version specified in the .nvmrc file (ensure node >= 18)
+   nvm use
+   npm install
+   ```
+
+4. Run the tests to make sure everything is working:
+
+   ```sh
+   npm test
+   ```
+
+5. Build the project:
+
+   ```sh
+   npm run build
+   ```
+
+If you're not sure where to start, check out our [good first issues](https://github.com/promptfoo/promptfoo/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22) or join our [Discord community](https://discord.gg/gHPS9jjfbs) for guidance.
+
+## Development Workflow
+
+1. Create a new branch for your feature or bug fix:
+
+   ```sh
+   git checkout -b feature/your-feature-name
+   ```
+
+2. We try to follow the [Conventional Commits](https://www.conventionalcommits.org/) specification. This is not required for feature branches. We merge all PRs into `main` with a squash merge and a conventional commit message.
+
+3. Push your branch to your fork:
+
+   ```sh
+   git push origin your-branch-name
+   ```
+
+4. Open a pull request against the `main` branch of the promptfoo repository.
+
+When opening a pull request:
+
+- Keep changes small and focused. Avoid mixing refactors with new features.
+- Ensure test coverage for new code or bug fixes.
+- Provide clear instructions on how to reproduce the problem or test the new feature.
+- Be responsive to feedback and be prepared to make changes if requested.
+- Ensure your tests are passing and your code is properly linted.
+
+Don't hesitate to ask for help. We're here to support you. If you're worried about whether your PR will be accepted, please talk to us first.
```

Comment:
If there is a specific Discord channel or preferred method of communications, this would be a useful place to include it.